lib/Frontend/ParseableInterfaceModuleLoader.cpp

```diff
@@ -620,6 +620,93 @@ class swift::ParseableInterfaceBuilder {
 
 namespace {
 
+/// Keeps track of the various reasons the module interface loader needed to
+/// fall back and rebuild a module from its interface.
+struct ModuleRebuildInfo {
+  enum class ModuleKind {
+    Normal,
+    Cached,
+    Forwarding,
+    Prebuilt
+  };
+  struct OutOfDateModule {
+    Optional<serialization::Status> serializationStatus;
+    ModuleKind kind;
+    SmallVector<std::string, 10> outOfDateDependencies;
+  };
+  llvm::StringMap<OutOfDateModule> outOfDateModules;
```

Comment:
The order here probably matters; should this be a vector instead?

Reply:
The issue is I'm using the module's path as a key into this dictionary, which allows the loader to just put breadcrumbs in. Since there are only ever at most 3 entries though, it's not expensive to just search through it...